carbon-design-system/carbon-addons-iot-react
# feat(tablecardeditform): allow edit form to be better customized

Closes #

**Summary**

- The TableCard needs to have additional abilities to edit the CardEditor panels to support our usecases

**Change List (commits, features, bugs, etc)**

- feat(CardEditFormContent): attach a class so styles can be more controlled and allow title, tooltip to be optional as well as ContentFormItemTitle
- feat(ContentFormItemTitle): allow title and tooltip to be optional
- feat(TableCardFormContent): attach a class so styles can be more controlled
- feat(card-edit-form.scss): remove incorrect --content styles (was not attached originally) and set to flex so it can be more controlled, same with the new table-card-form-content class
- feat(CardEditor): add new callback to dynamically provide valid dimensions instead of hardcoding them dashboard wide
- test(CardEditor.story): add an example of the table card being customized

**Acceptance Test (how to verify the PR)**

- Verify the new table card story of card editor correctly shows dimension values and updates the card when the DataType is changed

**Regression Test (how to make sure this PR doesn't break old functionality)**

- Verify the other CardEditor stories still match the old ones (no snapshot changes)



**Things to look for during review**
